### PR TITLE
CMake: handle potentially undefined value in list function

### LINKED ISSUE #3083: CMake configuration error due to logic in pc file creation
#### Example of problem

CMake configuration of proj 9.0.0 fails in some circumstances with `list sub-command REMOVE_ITEM requires two or more arguments.` coming from `configure_proj_pc` function.

<details><summary>Click to expand log</summary>

```
> cd '/Users/jenkins/w/BuildSingleReference@2/.conan/data/proj/9.0.0/_/_/build/629e0f9e00dc3db3465ab95c46401f4154870ebf' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Debug" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="apple-clang" -DCONAN_COMPILER_VERSION="12.0" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libc++" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/Users/jenkins/w/BuildSingleReference@2/.conan/data/proj/9.0.0/_/_/package/629e0f9e00dc3db3465ab95c46401f4154870ebf" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_MODULE_PATH="/Users/jenkins/w/BuildSingleReference@2/.conan/data/proj/9.0.0/_/_/build/629e0f9e00dc3db3465ab95c46401f4154870ebf" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DUSE_THREAD="True" -DBUILD_CCT="True" -DBUILD_CS2CS="True" -DBUILD_GEOD="True" -DBUILD_GIE="True" -DBUILD_PROJ="True" -DBUILD_PROJINFO="True" -DPROJ_DATA_SUBDIR="res" -DENABLE_TIFF="True" -DENABLE_CURL="True" -DBUILD_TESTING="False" -DENABLE_IPO="False" -DBUILD_PROJSYNC="True" -DNLOHMANN_JSON_ORIGIN="external" -DCMAKE_MACOSX_BUNDLE="False" -Wno-dev '/Users/jenkins/w/BuildSingleReference@2/.conan/data/proj/9.0.0/_/_/build/629e0f9e00dc3db3465ab95c46401f4154870ebf'
-----------------
-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Check for working C compiler: /Applications/conan/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/conan/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/conan/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/conan/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Using cmake targets configuration
-- Library sqlite3 found /Users/jenkins/w/BuildSingleReference@2/.conan/data/sqlite3/3.38.0/_/_/package/b60d76f22f7907f65b14f1c98cb7d91577ee8e2e/lib/libsqlite3.a
-- Library tiffxx found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiffxx.a
-- Library tiff found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiff.a
-- Library curl found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libcurl/7.80.0/_/_/package/90bd2a58d979c05796b8860544c6a35d876c5202/lib/libcurl.a
-- Library z found /Users/jenkins/w/BuildSingleReference@2/.conan/data/zlib/1.2.11/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libz.a
-- Library deflate found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libdeflate/1.9/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libdeflate.a
-- Library lzma found /Users/jenkins/w/BuildSingleReference@2/.conan/data/xz_utils/5.2.5/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/liblzma.a
-- Library jpeg found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libjpeg/9d/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libjpeg.a
-- Library jbig found /Users/jenkins/w/BuildSingleReference@2/.conan/data/jbig/20160605/_/_/package/8013bd69055ee988fd2ae37582b9c2f2a6a099e1/lib/libjbig.a
-- Library zstd found /Users/jenkins/w/BuildSingleReference@2/.conan/data/zstd/1.5.2/_/_/package/e63c4e1c6851042154c74d340413b965688640f2/lib/libzstd.a
-- Library webpdecoder found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdecoder.a
-- Library webpdemux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdemux.a
-- Library webpmux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpmux.a
-- Library webp found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebp.a
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Conan: C++ stdlib: libc++
-- Requiring C++11
-- Requiring C++11 - done
-- Requiring C99
-- Requiring C99 - done
-- Configuring PROJ:
-- PROJ_VERSION                   = 9.0.0
-- Looking for localeconv
-- Looking for localeconv - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for dladdr in dl
-- Looking for dladdr in dl - found
-- Looking for exp in m
-- Looking for exp in m - found
-- BUNDLEDIR                      = Applications/OSGEO
-- PROJ_INSTALL_NAME_DIR          = @executable_path/../../Library/Frameworks
-- FRAMEWORKDIR                   = Library/Frameworks
-- Conan: Using autogenerated Findnlohmann_json.cmake
-- Found nlohmann_json: 3.10.5 (found version "3.10.5") 
-- found nlohmann/json version 3.10.5
-- nlohmann/json: external
-- Found Sqlite3: /Users/jenkins/w/BuildSingleReference@2/.conan/data/sqlite3/3.38.0/_/_/package/b60d76f22f7907f65b14f1c98cb7d91577ee8e2e/lib/libsqlite3.a
-- Sqlite3 version: 3.38.0
-- Conan: Using autogenerated FindTIFF.cmake
-- Found TIFF: 4.3.0 (found version "4.3.0") 
-- Library tiffxx found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiffxx.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiffxx.a
-- Library tiff found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiff.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libtiff/4.3.0/_/_/package/ab8b9d41f7fcd447078fb62530e4baf9c7efd735/lib/libtiff.a
-- Conan: Using autogenerated FindZLIB.cmake
-- Found ZLIB: 1.2.11 (found version "1.2.11") 
-- Library z found /Users/jenkins/w/BuildSingleReference@2/.conan/data/zlib/1.2.11/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libz.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/zlib/1.2.11/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libz.a
-- Conan: Using autogenerated Findlibdeflate.cmake
-- Found libdeflate: 1.9 (found version "1.9") 
-- Library deflate found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libdeflate/1.9/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libdeflate.a
-- Skipping already existing target: CONAN_LIB::libdeflate_deflate
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libdeflate/1.9/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libdeflate.a
-- Conan: Using autogenerated FindLibLZMA.cmake
-- Found LibLZMA: 5.2.5 (found version "5.2.5") 
-- Library lzma found /Users/jenkins/w/BuildSingleReference@2/.conan/data/xz_utils/5.2.5/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/liblzma.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/xz_utils/5.2.5/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/liblzma.a
-- Conan: Using autogenerated FindJPEG.cmake
-- Found JPEG: 9d (found version "9d") 
-- Library jpeg found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libjpeg/9d/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libjpeg.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libjpeg/9d/_/_/package/d98fae1010d1fb9e7f79a1e8a72bbf129d8660a2/lib/libjpeg.a
-- Conan: Using autogenerated Findjbig.cmake
-- Found jbig: 20160605 (found version "20160605") 
-- Library jbig found /Users/jenkins/w/BuildSingleReference@2/.conan/data/jbig/20160605/_/_/package/8013bd69055ee988fd2ae37582b9c2f2a6a099e1/lib/libjbig.a
-- Skipping already existing target: CONAN_LIB::jbig_jbig
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/jbig/20160605/_/_/package/8013bd69055ee988fd2ae37582b9c2f2a6a099e1/lib/libjbig.a
-- Conan: Using autogenerated Findzstd.cmake
-- Found zstd: 1.5.2 (found version "1.5.2") 
-- Library zstd found /Users/jenkins/w/BuildSingleReference@2/.conan/data/zstd/1.5.2/_/_/package/e63c4e1c6851042154c74d340413b965688640f2/lib/libzstd.a
-- Skipping already existing target: CONAN_LIB::zstd_zstd
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/zstd/1.5.2/_/_/package/e63c4e1c6851042154c74d340413b965688640f2/lib/libzstd.a
-- Library zstd found /Users/jenkins/w/BuildSingleReference@2/.conan/data/zstd/1.5.2/_/_/package/e63c4e1c6851042154c74d340413b965688640f2/lib/libzstd.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/zstd/1.5.2/_/_/package/e63c4e1c6851042154c74d340413b965688640f2/lib/libzstd.a
-- Conan: Using autogenerated FindWebP.cmake
-- Found WebP: 1.2.2 (found version "1.2.2") 
-- Library webpdecoder found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdecoder.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdecoder.a
-- Library webpdemux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdemux.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdemux.a
-- Library webpmux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpmux.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpmux.a
-- Library webp found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebp.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebp.a
-- Library webp found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebp.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebp.a
-- Library webpmux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpmux.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpmux.a
-- Library webpdemux found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdemux.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdemux.a
-- Library webpdecoder found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdecoder.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libwebp/1.2.2/_/_/package/859f8116a362ef830363626989ec41f658f4e574/lib/libwebpdecoder.a
-- Conan: Using autogenerated FindCURL.cmake
-- Found CURL: 7.80.0 (found version "7.80.0") 
-- Library curl found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libcurl/7.80.0/_/_/package/90bd2a58d979c05796b8860544c6a35d876c5202/lib/libcurl.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libcurl/7.80.0/_/_/package/90bd2a58d979c05796b8860544c6a35d876c5202/lib/libcurl.a
-- Conan: Dependency ZLIB already found
-- Library curl found /Users/jenkins/w/BuildSingleReference@2/.conan/data/libcurl/7.80.0/_/_/package/90bd2a58d979c05796b8860544c6a35d876c5202/lib/libcurl.a
-- Found: /Users/jenkins/w/BuildSingleReference@2/.conan/data/libcurl/7.80.0/_/_/package/90bd2a58d979c05796b8860544c6a35d876c5202/lib/libcurl.a
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_PTHREAD_MUTEX_RECURSIVE_DEFN
-- Performing Test HAVE_PTHREAD_MUTEX_RECURSIVE_DEFN - Success
-- Testing disabled
-- Configuring proj library:
-- ENABLE_IPO                     = False
-- PROJ_CORE_TARGET_OUTPUT_NAME   = proj
-- BUILD_SHARED_LIBS              = OFF
-- PROJ_LIBRARIES                 = proj
-- PROJ: Configured 'dist' target
-- Configuring incomplete, errors occurred!
See also "/Users/jenkins/w/BuildSingleReference@2/.conan/data/proj/9.0.0/_/_/build/629e0f9e00dc3db3465ab95c46401f4154870ebf/CMakeFiles/CMakeOutput.log".
proj/9.0.0: 
CMake Error at source_subfolder/cmake/ProjUtilities.cmake:101 (list):
  list sub-command REMOVE_ITEM requires two or more arguments.
Call Stack (most recent call first):
  source_subfolder/CMakeLists.txt:329 (configure_proj_pc)
```

I've not been able to reproduce this error locally, only in CI of conan-center.

Proposed solution: test whether `CMAKE_C_IMPLICIT_LINK_LIBRARIES` variable is not empty before trying to call `list(REMOVE_ITEM ...)`.

</details>

#### Environment Information
 - PROJ version (`proj`): 9.0.0
 - Operation System Information: macOS

#### Installation method
 - built from source

### PR DESCRIPTION
**Pull request by mwtoews**
Thanks @SpaceIm confirming this fix!

Closes #3083

### CODE REVIEW

File: cmake/ProjUtilities.cmake [hunk 1]
```diff
@@ -98,7 +98,7 @@ function(configure_proj_pc)
     list(APPEND EXTRA_LIBS -lole32 -lshell32)
   else()
     set(cxx_libs "${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES}")
-    list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
+    list(REMOVE_ITEM cxx_libs "${CMAKE_C_IMPLICIT_LINK_LIBRARIES}")
```
Comment (dg0yt) [inline cmake/ProjUtilities.cmake]: This is the wrong fix.
We want to remove all individual list items in `CMAKE_C_IMPLICIT_LINK_LIBRARIES`, not the single item (or specific sequence) which is the concatenated value.
To work around the bug in old CMake versions, use a dummy item, i.e.
~~~cmake
list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES} CMake_3.19_quirk)
~~~

Cf. https://github.com/OSGeo/gdal/pull/5363

Comment (dg0yt) [reply cmake/ProjUtilities.cmake]: POC:
~~~cmake
set(list1 A B C)
set(list2 A C)
list(REMOVE_ITEM list1 "${list2}")
message(STATUS "${list1}")
list(REMOVE_ITEM list1 ${list2})
message(STATUS "${list1}")
list(REMOVE_ITEM list1 ${list3} CMake_3.19_quirk)
message(STATUS "${list1}")
~~~

CMake 3.19 output:
~~~
-- A;B;C
-- B
-- B
~~~

Comment (SpaceIm) [reply cmake/ProjUtilities.cmake]: what do you think of:

```cmake
if(CMAKE_C_IMPLICIT_LINK_LIBRARIES)
    list(REMOVE_ITEM cxx_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
endif()
```

Comment (dg0yt) [reply cmake/ProjUtilities.cmake]: This is what is done in GDAL. I dislike it because it is longer but still hides why it is needed at all.

Comment (mwtoews) [reply cmake/ProjUtilities.cmake]: Thanks @dg0yt for the useful example, I've modified this to the 1-liner, using a nonempty item at the end.

**Review by dg0yt — APPROVED** [review]

**Pull request merged**